File: gRPC poskus/clientGRPC.js
const grpc = require('@grpc/grpc-js');
const protoLoader = require('@grpc/proto-loader');
const PROTO_PATH = './nightsky.proto';

// Naloži gRPC definicijo iz .proto datoteke
const packageDefinition = protoLoader.loadSync(PROTO_PATH, {
    keepCase: true,
    longs: String,
    enums: String,
    defaults: true,
    oneofs: true
});
const nightskyProto = grpc.loadPackageDefinition(packageDefinition).NightSkyService;

// Ustvari klienta
const client = new nightskyProto('localhost:50051', grpc.credentials.createInsecure());

// Funkcija za pridobitev svetlobnega onesnaženja
function getLightPollution(location) {
    client.GetLightPollution({ location }, (err, response) => {
        if (err) {
            console.error("Error fetching light pollution data:", err);
        } else {
            console.log("Light Pollution Level:", response.level);
        }
    });
}

// Funkcija za pridobitev nočnih dogodkov
function getNightEvent() {
    client.GetNightEvent({}, (err, response) => {
        if (err) {
            console.error("Error fetching night event data:", err);
        } else {
            console.log("Night Event:", response.title);
            console.log("Event Details:", response.details);
        }
    });
}


// Funkcija za real-time posodobitve
function getRealTimeUpdates(location) {
    const stream = client.GetRealTimeUpdates({ location });
    stream.on('data', (data) => {
        console.log("Real-time data:", data.data);
    });
    stream.on('end', () => {
        console.log("Stream ended.");
    });
}

// Klic funkcij za testiranje
getLightPollution("Maribor");
getNightEvent();
getRealTimeUpdates("Celje");

File: gRPC poskus/clientGRPC.py
import grpc
import nightsky_pb2
import nightsky_pb2_grpc

# Kreiraj kanal i stub
channel = grpc.insecure_channel('localhost:50051')
stub = nightsky_pb2_grpc.NightSkyServiceStub(channel)

# Funkcija za pridobivanje svetlobnega onesnaženja
def get_light_pollution(location):
    request = nightsky_pb2.LightPollutionRequest(location=location)
    response = stub.GetLightPollution(request)
    print(f"Light Pollution Level in {location}: {response.level}")

# Funkcija za pridobivanje nočnih dogodkov
def get_night_event():
    request = nightsky_pb2.NightEventRequest()
    response = stub.GetNightEvent(request)
    print(f"Night Event: {response.title}")
    print(f"Event Details: {response.details}")

# Funkcija za real-time posodobitve
def get_real_time_updates(location):
    request = nightsky_pb2.StreamRequest(location=location)
    stream = stub.GetRealTimeUpdates(request)
    try:
        for update in stream:
            print(f"Real-time update: {update.data}")
    except grpc.RpcError as e:
        print(f"Stream ended with error: {e}")

# Klic funkcij za testiranje
if __name__ == "__main__":
    get_light_pollution("Maribor")
    get_night_event()
    get_real_time_updates("Celje")
